lib/Controller/WopiController.php

```diff
@@ -402,74 +424,51 @@ public function wopiPutFileRelative(string $documentId): JSONResponse {
 			return new JSONResponse([], Http::STATUS_FORBIDDEN);
 		}
 
-		// get owner and editor uid's
-		$owner = $res['owner'];
-		$editor = $res['editor'];
-
-		// Retrieve suggested target
-		$suggested = $this->request->getHeader('X-WOPI-SuggestedTarget');
-		$suggested = \iconv('utf-7', 'utf-8', $suggested);
-		
-		$file = $this->fileService->getFileHandle($fileId, $owner, $editor);
+		if ($res['editor'] && $res['editor'] !== '' && !($res['attributes'] & WOPI::ATTR_FEDERATED)) {
+			$file = $this->fileService->getFileHandle($res['fileid'], $res['owner'], $res['editor']);
+		} else {
+			$this->logger->warning('PutFileRelative: Unexpected call for function for anonymous access', ['app' => $this->appName]);
+			return new JSONResponse([], Http::STATUS_INTERNAL_SERVER_ERROR);
+		}
 
 		if (!$file) {
-			$this->logger->warning('PutFileRelative: could not retrieve file', ['app' => $this->appName]);
+			$this->logger->warning('PutFileRelative: Could not retrieve file', ['app' => $this->appName]);
 			return new JSONResponse([], Http::STATUS_NOT_FOUND);
 		}
 
-		$path = '';
+		// Retrieve suggested target
+		$suggested = $this->request->getHeader('X-WOPI-SuggestedTarget');
+		$suggested = \iconv('utf-7', 'utf-8', $suggested);
```

Comment:
This might need a comment. It could be useful to know why we need to convert the charset because it isn't a common operation.

Reply:
I would love to know too why this was done years back 😆

Reply:
I will add a comment